test/java/org/opendatakit/briefcase/util/StringUtilsTest.java

```diff
@@ -1,53 +1,120 @@
-    Assert.assertEquals("abcdef   ", output);
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+        {"Null character \\u0000", "hello\u0000world", "helloworld"},
+        {"Control character \\n", "hello\nworld", "helloworld"},
+        {"Restricted code /", "hello/world", "helloworld"},
+        {"Restricted code ?", "hello?world", "helloworld"},
+        {"Restricted code <", "hello<world", "helloworld"},
+        {"Restricted code >", "hello>world", "helloworld"},
+        {"Restricted code \\", "hello\\world", "helloworld"},
+        {"Restricted code *", "hello*world", "helloworld"},
+        {"Restricted code |", "hello|world", "helloworld"},
+        {"Restricted code \"", "hello\"world", "helloworld"},
+        {"Restricted code :", "hello:world", "helloworld"},
+        {"Restricted prefix .", ".hello world", ".hello world"},
+        {"Restricted prefix ..", "..hello world", "..hello world"},
+        {"Restricted prefix \" \"", " hello world", " hello world"},
+        {"Restricted prefix \"  \"", "  hello world", "  hello world"},
+        {"Restricted suffix .", "hello world.", "hello world"},
+        {"Restricted suffix ..", "hello world..", "hello world"},
+        {"Restricted suffix \" \"", "hello world ", "hello world"},
+        {"Restricted suffix \"  \"", "hello world  ", "hello world"},
+        {"Relative path .", ".", ""},
+        {"Relative path ..", "..", ""},
+        {"Relative path ./", "./", ""},
+        {"Relative path ../", "../", ""},
+        {"Relative path /..", "/..", ""},
+        {"Relative path /../", "/../", ""},
+        {"Reserved Windows filename CON", "CON", ""},
+        {"Reserved Windows filename PRN", "PRN", ""},
+        {"Reserved Windows filename AUX", "AUX", ""},
+        {"Reserved Windows filename NUL", "NUL", ""},
+        {"Reserved Windows filename COM1", "COM1", ""},
+        {"Reserved Windows filename COM2", "COM2", ""},
+        {"Reserved Windows filename COM3", "COM3", ""},
+        {"Reserved Windows filename COM4", "COM4", ""},
+        {"Reserved Windows filename COM5", "COM5", ""},
+        {"Reserved Windows filename COM6", "COM6", ""},
+        {"Reserved Windows filename COM7", "COM7", ""},
+        {"Reserved Windows filename COM8", "COM8", ""},
+        {"Reserved Windows filename COM9", "COM9", ""},
+        {"Reserved Windows filename LPT1", "LPT1", ""},
+        {"Reserved Windows filename LPT2", "LPT2", ""},
+        {"Reserved Windows filename LPT3", "LPT3", ""},
+        {"Reserved Windows filename LPT4", "LPT4", ""},
+        {"Reserved Windows filename LPT5", "LPT5", ""},
+        {"Reserved Windows filename LPT6", "LPT6", ""},
+        {"Reserved Windows filename LPT7", "LPT7", ""},
+        {"Reserved Windows filename LPT8", "LPT8", ""},
+        {"Reserved Windows filename LPT9", "LPT9", ""},
+        {"Reserved Windows filename con", "con", ""},
```

Comment:
I’d rather see these lower-case versions generated rather than duplicated, but I suppose if the items aren’t expected to change much or at all, there’s little risk someone will change one and not the other.

Reply:
Thanks for the suggestion! It's better :)